Keas.Mvc/ClientApp/components/Equipment/EquipmentBigFixSearchId.tsx

```diff
@@ -0,0 +1,280 @@
+                            });
+                        }}
+                    />
+                </>
+            );
+        } else if (this.state.selectedFeild === "Id") {
+            return <Input type="text" name="Id" id="computer Id" placeholder="Enter Computer Id" />;
+        } else if (this.state.selectedFeild === "Company") {
+            return (
+                <Input
+                    type="text"
+                    name="Company"
+                    id="computer Company"
+                    placeholder="Enter Computer Company"
+                />
+            );
+        }
+    };
+
+    private _renderNameTable = () => {
+        if (!this.state.isFetched) {
+            return null;
+        }
+
+        if (!this.state.isValidSearch) {
+            return (
+                <p className="text-center text-danger">
+                    No computer found by this name, please try again.
+                </p>
+            );
+        }
+        return (
+            <Table>
+                <tbody>
+                    {this.state.listOfComputers.map(computer => {
+                        return (
+                            <tr
+                                className="bigfix-info"
+                                onClick={() =>
+                                    this.props.addBigFixId("systemManagementId", computer.id)
+                                }
+                                key={computer.id}
+                            >
+                                <td>{computer.name}</td>
+                            </tr>
+                        );
+                    })}
+                </tbody>
+            </Table>
+        );
+    };
+
+    private _renderSearchButton = () => {
+        if (this.state.isSearching) {
+            return (
+                <Button color="primary" disabled={false}>
+                    {" "}
+                    <i className="fas fa-lg fa-spinner fa-pulse" />
+                </Button>
+            );
+        } else {
+            return (
+                <Button
+                    color="primary"
+                    disabled={false}
+                    onClick={() => {
+                        this.setState({ isSearching: true, isFetched: false, isValidSearch: true });
+                        this._getComputersBySearch(
+                            this.state.selectedFeild,
+                            this.state.valueToBeSearched
+                        );
+                    }}
+                >
+                    Search!
+                </Button>
+            );
+        }
+    };
+
+    private _getComputersBySearch = async (field: string, value: string) => {
+        const response = await fetch(
+            `/api/${this.context.team.slug}/equipment/GetComputersBySearch?field=${field}&value=${value}`,
+            {
+                method: "GET"
+            }
+        );
+
+        if (!response.ok) {
```

Comment:
instead of doing `!response.ok`, put your fetch in a try/catch and handle this in the catch. if you look in `api.ts` you can see that it does this for you